scripts/data/client.py

```diff
@@ -16,32 +15,51 @@
 from format_args import format_args
 import logging
 from logging.handlers import TimedRotatingFileHandler
+import signal
+import sys
 
 logger = logging.getLogger(__name__)
 
 # Constants
 MAX_WEIGHT_LIMIT = 8000  # Total weight limit for all jobs
 THREAD_POOL_SIZE = os.cpu_count()  # Number of threads for processing
 QUEUE_MAX_SIZE = THREAD_POOL_SIZE * 2  # Maximum size of the job queue
-
 BASE_DIR = Path(".client_cache")
 
 # Shared state variables
 current_weight = 0
 weight_lock = threading.Condition()
 job_queue = queue.Queue(maxsize=QUEUE_MAX_SIZE)
 
+shutdown_event = threading.Event()
```

Comment:
does this need to be an event? No one waits for it.

Reply:
Yes it does, because it is checked in multiple threads and also a part of the shutdown signal system